docs/uploading_and_publishing/explanation_how_the_processing_works.md

```diff
@@ -0,0 +1,158 @@
+    `workflow` is an old version of the workflow section, while `workflow2` is the newest one. `workflow` is still present in some of the older entries, so 
+    that until the full NOMAD repository and its entries are reprocessed with the new section `workflow2`, we need to maintain the old section.
+    This is an usual practice in our development process, especially for sections which needs refactoring.
+
+
+### Parsing {#parsing}
+
+A parser is a Python script which reads the code-specific mainfile and auxiliary files and populates the `run` and `workflow2` sections, and all the sub-sections and quantities defined therein. We explain them more in detail in [Writing a parser plugin](../writing_a_parser_plugin/parser_plugin_overview.md). 
+
+Parsers can be added to NOMAD as _plugins_ and are divided in a set of Github sub-projects under the [main NOMAD repository](https://github.com/nomad-coe/nomad). The path to these projects can be found in `/dependencies/parsers/<project>` and they are:
+
+- [atomistic](https://github.com/nomad-coe/atomistic-parsers): parsers for classical molecular and atomistic simulations, e.g., from Gromacs, Lammps, etc.
+- [database](https://github.com/nomad-coe/database-parsers): parsers for various databases, e.g., OpenKim.
+- [eelsdb](https://github.com/nomad-coe/nomad-parser-eelsdb): parser for the [EELS database](https://eelsdb.eu/). To be integrated in the database project.
+- [electronic](https://github.com/nomad-coe/electronic-parsers): parsers for electronic structure calculations, e.g., VASP, FHI-aims, etc.
+- [nexus](https://github.com/nomad-coe/nomad-parser-nexus): parsers for combining various instrument output formats and electronic lab notebooks.
+- [workflow](https://github.com/nomad-coe/workflow-parsers): parsers for task managers and workflow schedulers.
+
+!!! tip "External contributions"
+    We always welcome external contributions for new codes and parsers in our repositories. Furthermore, we are always happy to hear feedback and implement new features
+    into our parsers. Please, feel free to reach to us by email or by opening an issue in one of the Github repositories and we will promptly reach out back to help you!
+
+
+### Normalizing
+
+After the parsing populates `run` and `workflow2` sections, a new layer of Python scripts is executed on top of the processed NOMAD metadata. This has two main purposes: one, normalize or _homogenize_ certain metadata parsed from different codes, and two, populate the `results` section. We explain them more in detail in [Normalizing the parsed metadata](../normalizing_the_parsed_metadata/intro.md)
+
+The development of normalizers in NOMAD is usually an internal process, as it requires across-codes and across-areas robustness, because these are applied to both computational and experimental metadata. Furthermore, it is linked with the indexing and searching of data (see next section).
```

Comment:
> I think we need to be careful here. What is the purpose of this section?

Agreed, this is deviating from the objective of this page. Maybe it should then be renamed to _NOMAD software structure_. To be honest, if normalizers aren't relevant to the user, then don't mention them. They are only relevant once you start contributing code. Moreover, the distinction between when something goes into parser or normalizer isn't clean cut at all.

> You say that this is an internal process

I guess because you need permissions to the base project, which is a bit more sensitive? Depends on Markus or the PIs.

Regardless, the parser-normalizer separation needs overhauling. The way to go in the future is design functions with well-defined interfaces that construct the (sub)sections. Any data-manipulation on their end could be considered "normalizing", and the distinction will be much cleaner then. The question for contributing will be "where do they go?"

Reply:
I can delete this phrase, and later have it in mind when writing about "Normalizing the parsed metadata".